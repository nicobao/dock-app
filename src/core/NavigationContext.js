import React, {useEffect, useState} from 'react';

export const NavigationContext = React.createContext({currentTab: 'tokens'});
export const withNavigationContext = WrappedComponent => props => {
  const [currentTab, setCurrentTab] = useState('tokens');
  const {route} = props;

  useEffect(() => {
    if (route && route?.params?.currentTab) {
      setCurrentTab(route.params.currentTab);
    }
  }, [route]);

  return (
    <NavigationContext.Provider
      value={{
        currentTab,
      }}>
      <WrappedComponent {...props} />
    </NavigationContext.Provider>
  );
};
