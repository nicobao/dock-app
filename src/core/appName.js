export const app = {
  name: 'Dock Wallet',
};
