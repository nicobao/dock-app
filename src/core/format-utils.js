import {
  formatCurrency,
  formatAddress,
  formatDockAmount,
  getPlainDockAmount,
  formatDate,
} from '@docknetwork/wallet-sdk-core/lib/core/format-utils';

export {
  formatCurrency,
  formatAddress,
  formatDockAmount,
  getPlainDockAmount,
  formatDate,
};
