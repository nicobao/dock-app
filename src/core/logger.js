const winston = require('winston');

export const Logger = winston.createLogger({
  format: winston.format.simple(),
  level: 'info',
  transports: [new winston.transports.Console()],
});
