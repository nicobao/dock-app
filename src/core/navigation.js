import {createNavigationContainerRef} from '@react-navigation/native';

export const navigationRef = createNavigationContainerRef();

export function navigate(name, params) {
  if (navigationRef.isReady()) {
    navigationRef.navigate(name, params);
  }
}

export function navigateBack() {
  if (navigationRef.isReady()) {
    navigationRef.goBack();
  }
}
