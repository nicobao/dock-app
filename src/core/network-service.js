import axios from 'axios';
import {RequestLogger} from '@docknetwork/wallet-sdk-request-logger/lib/request-logger';
const instance = axios.create({});
import store from '../core/redux-store';
instance.interceptors.response.use(
  function (response) {
    const {url, method, headers, data} = response.config;
    const {status, data: payload} = response;

    const shouldLogRequest =
      store.getState()?.app?.features?.shouldLogRequest || false;
    if (shouldLogRequest) {
      RequestLogger.logRequest({
        url,
        method,
        headers,
        body: data,
        response: payload,
        status,
      });
    }
    return response;
  },
  function (error) {
    const shouldLogRequest =
      store.getState()?.app?.features?.shouldLogRequest || false;
    const response = error.response;
    const {status, data: payload} = response;

    const {url, method, headers, data} = response.config;
    if (shouldLogRequest) {
      RequestLogger.logRequest({
        url,
        method,
        headers,
        body: data,
        response: payload,
        status,
      });
    }
    return Promise.reject(error);
  },
);

export default instance;
