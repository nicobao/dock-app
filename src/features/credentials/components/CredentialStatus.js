import {NBox, Typography} from '../../../design-system';
import {HStack} from 'native-base';
import React from 'react';

import {useGetCredentialStatus} from '../hooks/credentialHooks';

export function CredentialStatus({credential}) {
  const {statusData} = useGetCredentialStatus({credential});

  return (
    <HStack ml={3}>
      <NBox mt={0.5}>{statusData.icon}</NBox>
      <Typography
        color={statusData.color}
        ml={1}
        variant={'credentialIssuanceDate'}>
        {statusData.message}
      </Typography>
    </HStack>
  );
}
