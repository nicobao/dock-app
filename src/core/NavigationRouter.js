import {NavigationContainer} from '@react-navigation/native';

import React, {useEffect} from 'react';
import {Linking, Platform} from 'react-native';

import {navigationRef} from './navigation';

import DeepLinking from 'react-native-deep-linking';

import {authenticationSelectors} from '../features/unlock-wallet/unlock-wallet-slice';
import {useSelector} from 'react-redux';
import {useDeepLink} from './hooks/deepLinkHooks';
import {UnAuthNavigationStackScreen} from './navigation/unauthenticated-stack';
import {AuthNavigationStackScreen} from './navigation/authenticated-stack';

const forFade = ({current, closing}) => ({
  cardStyle: {
    opacity: current.progress,
  },
});

const screenOptions = {
  headerShown: false,
};

if (Platform.OS === 'android') {
  screenOptions.cardStyleInterpolator = forFade;
}

export function NavigationRouter() {
  const isLoggedIn = useSelector(authenticationSelectors.isLoggedIn);
  const {gotoScreenDeepLink} = useDeepLink({isLoggedIn});

  useEffect(() => {
    const getAsyncURL = async () => {
      const initialUrl = await Linking.getInitialURL();
      if (initialUrl !== undefined && initialUrl != null) {
        gotoScreenDeepLink({url: initialUrl});
      }
    };

    getAsyncURL();
  }, [gotoScreenDeepLink]);

  useEffect(() => {
    DeepLinking.addScheme('dockwallet://');
    Linking.addEventListener('url', gotoScreenDeepLink);

    return () => {
      Linking.removeEventListener('url', gotoScreenDeepLink);
    };
  }, [gotoScreenDeepLink]);

  return (
    <NavigationContainer ref={navigationRef}>
      {isLoggedIn ? (
        <AuthNavigationStackScreen />
      ) : (
        <UnAuthNavigationStackScreen />
      )}
    </NavigationContainer>
  );
}
