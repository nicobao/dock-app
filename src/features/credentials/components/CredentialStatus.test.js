import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {CredentialStatus} from './CredentialStatus';
import {renderAppProviders} from '../../../core/test-utils';

const mockStore = configureMockStore();

describe('CredentialsScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const credential = {
      '@context': ['https://www.w3.org/2018/credentials/v1'],
      id: 'https://creds.dock.io/8e02c35ae370b02f47d7faaf41cb1386768fc75c9fca7caa6bb389dbe61260eb',
      type: ['VerifiableCredential', 'UniversityDegreeCredential'],
      credentialSubject: {},
      issuanceDate: '2022-06-27T12:08:30.675Z',
      expirationDate: '2039-06-26T23:00:00.000Z',
      issuer: {
        name: 'John Doe',
        description: '',
        logo: '',
        id: 'did:dock:5CJaTP2eGCLf5ZNPUXYbWxUvJQMTseKfc4hi8WVBC1K8eW9N',
      },
    };
    const wrapper = shallow(
      renderAppProviders(<CredentialStatus credential={credential} />),
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
