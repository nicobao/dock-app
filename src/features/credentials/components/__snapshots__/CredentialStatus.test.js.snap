// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialsScreen should render correctly when expired 1`] = `
<NativeBaseProvider
  initialWindowMetrics={
    Object {
      "frame": Object {
        "height": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      },
      "insets": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <CredentialStatus
    credential={
      Object {
        "id": 1,
        "status": 2,
      }
    }
  />
</NativeBaseProvider>
`;

exports[`CredentialsScreen should render correctly when invalid 1`] = `
<NativeBaseProvider
  initialWindowMetrics={
    Object {
      "frame": Object {
        "height": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      },
      "insets": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <CredentialStatus
    credential={
      Object {
        "id": 1,
        "status": 1,
      }
    }
  />
</NativeBaseProvider>
`;

exports[`CredentialsScreen should render correctly when revoked 1`] = `
<NativeBaseProvider
  initialWindowMetrics={
    Object {
      "frame": Object {
        "height": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      },
      "insets": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <CredentialStatus
    credential={
      Object {
        "id": 1,
        "status": 4,
      }
    }
  />
</NativeBaseProvider>
`;

exports[`CredentialsScreen should render correctly when verified 1`] = `
<NativeBaseProvider
  initialWindowMetrics={
    Object {
      "frame": Object {
        "height": 0,
        "width": 0,
        "x": 0,
        "y": 0,
      },
      "insets": Object {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    }
  }
>
  <CredentialStatus
    credential={
      Object {
        "id": 1,
        "status": 3,
      }
    }
  />
</NativeBaseProvider>
`;
